src/test/java/com/bvan/vlad/MessageServiceTest.java

```diff
@@ -0,0 +1,149 @@
+package com.bvan.vlad;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import org.junit.jupiter.api.Test;
+
+public class MessageServiceTest {
+
+    MessageService messageService = new MessageService();
+
+    @Test
+    public void lastMessageDate_happyPath_withBigData() {
+        // given
+        List<List<Message>> batches = new ArrayList<>();
+        for (int i = 1; i <= 10; i++) {
+            batches.add(getRandomBatchForBigData());
+        }
+        Message lastMessage = new Message("id", "last", Instant.now().plusSeconds(1_000_001));
+        batches.get(0).add(lastMessage);
+        // when
+        Instant instant = messageService.lastMessageDate(batches);
+        // then
+        assertEquals(lastMessage.getCreatedDate(), instant);
+    }
+
+    @Test
+    public void lastMessageDate_happyPath() {
+        // given
+        List<List<Message>> batches = new ArrayList<>();
+        for (int i = 1; i <= 11; i++) {
+            batches.add(getRandomBatch());
+        }
+        Message lastMessage = new Message("id", "last", Instant.now().plusSeconds(86400));
+        batches.get(0).add(lastMessage);
+        // when
+        Instant instant = messageService.lastMessageDate(batches);
+        // then
+        assertEquals(lastMessage.getCreatedDate(), instant);
+    }
+
+    @Test
+    public void lastMessageDate_whenBatchesIsEmpty_thenCannotReturnInstant() {
+        // given
+        List<List<Message>> batches = new ArrayList<>();
+        // when
+        assertThrows(NoSuchElementException.class, () -> messageService.lastMessageDate(batches));
+        // then throw exception
+
+    }
+
+    @Test
+    public void lastMessageDate_whenListsMessagesInBatchesIsEmpty_thenCannotReturnInstant() {
```

Comment:
What is happen if a single batch is empty?

Reply:
nothing